apps/sms/js/thread_list_ui.js

```diff
@@ -530,7 +490,64 @@ var ThreadListUI = {
         text: 'delete',
         className: 'danger'
       }
-    ).then(performDeletion.bind(this));
+    ).then(performDeletion.bind(this, selected));
+  },
+
+  deleteThreadDraftUI:
+    function thlui_deleteThreadDraftUI(selected, draftIds, threadIds) {
+    if(draftIds) {
+      draftIds.forEach(function(threadId) {
+        ThreadListUI.removeThread(threadId);
+      });
+    }
+    if (!threadIds.length) {
+      ThreadListUI.exitEditMode();
+    } else if(threadIds) {
+      threadIds.forEach(function(threadId) {
+        ThreadListUI.deleteThread(threadId);
+      });
+    }
+    ThreadListUI.deleteThreadDraft(selected, draftIds, threadIds);
+  },
+
+  deleteThreadDraft:
+    function thlui_deleteThreadDraft(selected, draftIds, threadIds) {
+    var threadCountToDelete = 0, messageIdsToDelete = [];
+
+    if(draftIds) {
+      draftIds.forEach(function(threadId) {
+        Drafts.delete(Drafts.get(threadId));
+      });
+    }
+    if (threadIds.length !== selected.length) {
+      Drafts.store();
+    }
+
+    function onAllThreadMessagesRetrieved() {
+      if (!--threadCountToDelete) {
+        MessageManager.deleteMessages(messageIdsToDelete);
+        ThreadListUI.exitEditMode();
+      }
+    }
+
+    function onThreadMessageRetrieved(message) {
+      messageIdsToDelete.push(message.id);
+      return true;
+    }
+    threadCountToDelete = threadIds.length;
+    threadIds.forEach(function(threadId) {
+      MessageManager.getMessages({
+        // Filter and request all messages with this threadId
+        filter: { threadId: threadId },
+        each: onThreadMessageRetrieved,
+        end: onAllThreadMessagesRetrieved
+      });
+    });
+  },
+
+  exitEditMode: function thlui_exitEditMode() {
+    ThreadListUI.cancelEdit();
+    WaitingScreen.hide();
```

Comment:
actually, you can move `WaitingScreen.hide()` to `cancelEdit` and just use `cancelEdit` everywhere.